File: client/python/gamium/gamium/options/move_player_options.py
from typing import Optional
from gamium.protocol.generated.Packets.Actions.MovePlayerBy import MovePlayerBy


class MovePlayerOptions:
    def __init__(
        self,
        by: int = MovePlayerBy.KeyPress,
        epsilon: float = 0.6,
        check_height: bool = False,
    ):
        self.by = by
        self.epsilon = epsilon
        self.check_height = check_height

File: client/typescript/gamium/src/common/options/move-player-options.ts
import { MovePlayerBy } from '../protocols/generated';

export interface MovePlayerOptions {
  by: MovePlayerBy;
  epsilon: number;
  checkHeight: boolean;
}

export function DefaultMovePlayerOptions(): MovePlayerOptions {
  return {
    by: MovePlayerBy.KeyPress,
    epsilon: 0.6,
    checkHeight: false,
  };
}
